R/shiny.R

```diff
@@ -58,11 +64,25 @@ renderDataTable = function(expr, env = parent.frame(), quoted = FALSE, ...) {
     instance = exprFunc()
     if (!all(c('datatables', 'htmlwidget') %in% class(instance))) {
       instance = datatable(instance, ...)
+    } else {
+      if (length(list(...)) != 0) {
+        warning("renderDataTable ignores ... arguments when expr yields a datatable object; see ?renderDataTable")
+      }
     }
 
-    if (is.function(hook <- instance$preRenderHook))
-      instance = hook(instance, currentSession, currentOutputName)
-    instance$preRenderHook = NULL
+    # in the server mode, we should not store the full data in JSON
+    if (server) {
+      origData = instance[['x']][['data']]
+      instance$x$data = NULL
```

Comment:
Oh, that's true. If the items in `instance` are fully under your control, it shouldn't be a problem as long as you do development with `options(warnPartialMatchDollar = TRUE)`. It's probably a good idea in general to also enable `warnPartialMatchArgs` and `warnPartialMatchAttr`.